src/mlpack/methods/pca/pca_impl.hpp

```diff
@@ -74,6 +74,23 @@ void PCA<DecompositionPolicy>::Apply(const arma::mat& data,
   arma::mat eigvec;
   Apply(data, transformedData, eigVal, eigvec);
 }
+    
+/**
+ * This is another Overload of apply with only 2 parameteres(data & transformed data)
```

Comment:
Do you mind to go with `Apply Principal Component Analysis to the provided data set.` instead here, a user doesn't necessarily know or care about another overload or if the parameters are stored internally.

Reply:
@zoq  A similar overload is already here on which you commented , it creates `arma :: mat eigvec` on its own and user doesn't need to pass it. Similarly in my added overload user doesn't need to pass even the `arma :: vec eigval` and both these overloads do the same thing except storing of parameters. User may or may not be aware so either we should keep both overloads or the only the one which is added in this PR because **later one is creating both parameters on its own(more easy for the user)** and there is no use of creating one parameter and passing one by user(as done  [here](https://github.com/mlpack/mlpack/blob/56ac3d219b9dc7281ebee5b2385302943070e180/src/mlpack/methods/pca/pca_impl.hpp#L70)) . 
So alternatively I can replace that overload(link mentioned in above line) with this one(added in this PR).

Reply:
I was just referencing the method description, if a user uses a specific interface we don't have to tell him we create parameter x,y for you. Does this make sense?

Reply:
@zoq okay so i will change the method description